CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataDouble.java

```diff
@@ -17,7 +17,7 @@ public DataDouble(double value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a double to a bool");
+        return value != 0.0;
```

Comment:
I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1.0`